sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java

```diff
@@ -386,6 +387,8 @@ protected void finalize() throws Throwable {
             currentReader.close();
           } catch (IOException e) {
             LOG.error("Failed to close BoundedReader due to failure processing bundle.", e);
+          } finally {
+            currentReader = null;
```

Comment:
spotbugs seems not happy about this

Reply:
Thanks! Fixed spotbugs issue about setting field to null in finalize() method, which is not recommended as it is being GC'd anyway.